backend/routes/rankings.js

```diff
@@ -1,52 +1,81 @@
 import express from "express";
 import { Follow } from "../models/follow.js";
 import { Stock } from "../models/stock.js";
+import { Like } from "../models/like.js";
+import { Dislike } from "../models/dislike.js";
+import { compareStocksByPoints } from "../utils.js";
+
+const FOLLOW_POINTS = 25;
+const LIKE_POINTS = 20;
+const DISLIKE_POINTS = -20;
 
 const router = express.Router();
 
 router.get("/ranking", async (req, res) => {
     const user = req.session.user;
 
     try {
-        const follows = await Follow.findAll({
-            where: { UserId: user.id },
-            include: [{ model: Stock }]
-        });
-
-        const stocksYouFollow = follows.length === 0 ? [] : follows.map((follow) => follow.Stock);
-
-        const stocksInDatabase = await Stock.findAll();
+        const [stocksInDatabase, follows, likes, dislikes] = await Promise.all([
+            Stock.findAll(),
+            Follow.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Like.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Dislike.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            })
+        ]);
 
         if (stocksInDatabase.length < 10) {
             res.status(422).json({
                 error: "You need at least 10 stocks in the database to access ranking."
             });
         }
 
-        const randomStocksNeeded = 10 - stocksYouFollow.length;
+        const stocksYouFollow = follows.map((follow) => follow.Stock);
+        const stocksYouLike = likes.map((like) => like.Stock);
+        const stocksYouDislike = dislikes.map((dislike) => dislike.Stock);
 
```

Comment:
since you're already iterating over the entire lists of `follows`, `likes`, and `dislikes`, you could consider creating these as a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)